src/core/src/OmniWebMapServiceRasterOverlay.cpp

```diff
@@ -0,0 +1,102 @@
+#include "cesium/omniverse/OmniWebMapServiceRasterOverlay.h"
+
+#include "cesium/omniverse/AssetRegistry.h"
+#include "cesium/omniverse/Context.h"
+#include "cesium/omniverse/GltfUtil.h"
+#include "cesium/omniverse/Logger.h"
+#include "cesium/omniverse/OmniGeoreference.h"
+#include "cesium/omniverse/UsdUtil.h"
+
+#include <CesiumRasterOverlays/WebMapServiceRasterOverlay.h>
+#include <CesiumUsdSchemas/webMapServiceRasterOverlay.h>
+
+namespace cesium::omniverse {
+
+OmniWebMapServiceRasterOverlay::OmniWebMapServiceRasterOverlay(Context* pContext, const pxr::SdfPath& path)
+    : OmniRasterOverlay(pContext, path) {
+    reload();
+}
+
+CesiumRasterOverlays::RasterOverlay* OmniWebMapServiceRasterOverlay::getRasterOverlay() const {
+    return _pWebMapServiceRasterOverlay.get();
+}
+
+std::string OmniWebMapServiceRasterOverlay::getBaseUrl() const {
+    auto cesiumWebMapServiceRasterOverlay =
+        UsdUtil::getCesiumWebMapServiceRasterOverlay(_pContext->getUsdStage(), _path);
+    std::string baseUrl;
+    cesiumWebMapServiceRasterOverlay.GetBaseUrlAttr().Get(&baseUrl);
+    return baseUrl;
+}
+
+int OmniWebMapServiceRasterOverlay::getMinimumLevel() const {
+    auto cesiumWebMapServiceRasterOverlay =
+        UsdUtil::getCesiumWebMapServiceRasterOverlay(_pContext->getUsdStage(), _path);
+    int val;
+    cesiumWebMapServiceRasterOverlay.GetMinimumLevelAttr().Get(&val);
+    return val;
+}
+
+int OmniWebMapServiceRasterOverlay::getMaximumLevel() const {
+    auto cesiumWebMapServiceRasterOverlay =
+        UsdUtil::getCesiumWebMapServiceRasterOverlay(_pContext->getUsdStage(), _path);
+    int val;
+    cesiumWebMapServiceRasterOverlay.GetMaximumLevelAttr().Get(&val);
+    return val;
+}
+
+int OmniWebMapServiceRasterOverlay::getTileWidth() const {
+    auto cesiumWebMapServiceRasterOverlay =
+        UsdUtil::getCesiumWebMapServiceRasterOverlay(_pContext->getUsdStage(), _path);
+    int val;
+    cesiumWebMapServiceRasterOverlay.GetTileWidthAttr().Get(&val);
+    return val;
+}
+
+int OmniWebMapServiceRasterOverlay::getTileHeight() const {
+    auto cesiumWebMapServiceRasterOverlay =
+        UsdUtil::getCesiumWebMapServiceRasterOverlay(_pContext->getUsdStage(), _path);
+    int val;
+    cesiumWebMapServiceRasterOverlay.GetTileHeightAttr().Get(&val);
+    return val;
+}
+
+std::string OmniWebMapServiceRasterOverlay::getLayers() const {
+    auto cesiumWebMapServiceRasterOverlay =
+        UsdUtil::getCesiumWebMapServiceRasterOverlay(_pContext->getUsdStage(), _path);
+    std::string val;
+    cesiumWebMapServiceRasterOverlay.GetLayersAttr().Get(&val);
+    return val;
+}
+
+void OmniWebMapServiceRasterOverlay::reload() {
+    const auto rasterOverlayName = UsdUtil::getName(_pContext->getUsdStage(), _path);
+
+    CesiumRasterOverlays::RasterOverlayOptions options;
+    options.showCreditsOnScreen = getShowCreditsOnScreen();
+    // options.ktx2TranscodeTargets = GltfUtil::getKtx2TranscodeTargets();
```

Comment:
Any reason this is commented out?

Reply:
Shouldn't be there. I missed it looking at commits. WMS textures weren't loading so I did a quick check to see if it could possibly be transcoding.